primitives/arithmetic/src/per_things.rs

```diff
@@ -309,6 +309,30 @@ where
 	rem_mul_div_inner.into()
 }
 
+macro_rules! implement_per_thing_with_permill {
```

Comment:
```suggestion
macro_rules! implement_per_thing_with_perthousand {
```

Or else I don't get the naming intention here.